### ISSUE TITLE
java.lang.IllegalArgumentException: Argument for @NotNull parameter 'dir' of zielu/gittoolbox/cache/VirtualFileRepoCacheImpl.getRepoForDir must not be null

### ISSUE THREAD

**Issue by intr** [zielu/GitToolBox#110]
Exception happened when I switched Git branch.

```
java.lang.IllegalArgumentException: Argument for @NotNull parameter 'dir' of zielu/gittoolbox/cache/VirtualFileRepoCacheImpl.getRepoForDir must not be null
	at zielu.gittoolbox.cache.VirtualFileRepoCacheImpl.$$$reportNull$$$0(VirtualFileRepoCacheImpl.java)
	at zielu.gittoolbox.cache.VirtualFileRepoCacheImpl.getRepoForDir(VirtualFileRepoCacheImpl.java)
	at zielu.gittoolbox.cache.VirtualFileRepoCache.getRepoForFile(VirtualFileRepoCache.java:29)
	at zielu.gittoolbox.blame.BlameServiceImpl.currentRepoRevision(BlameServiceImpl.java:219)
	at zielu.gittoolbox.blame.BlameServiceImpl.getCurrentLineBlameInternal(BlameServiceImpl.java:140)
	at zielu.gittoolbox.blame.BlameServiceImpl.getCurrentLineBlameInternal(BlameServiceImpl.java:131)
	at zielu.gittoolbox.blame.BlameServiceImpl.lambda$getCurrentLineBlame$1(BlameServiceImpl.java:97)
	at com.codahale.metrics.Timer.timeSupplier(Timer.java:133)
	at zielu.gittoolbox.blame.BlameServiceImpl.getCurrentLineBlame(BlameServiceImpl.java:97)
	at zielu.gittoolbox.ui.statusbar.BlameStatusWidget.fileChanged(BlameStatusWidget.java:279)
	at zielu.gittoolbox.ui.statusbar.BlameStatusWidget.fileOpened(BlameStatusWidget.java:247)
	at sun.reflect.GeneratedMethodAccessor68.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:117)
	at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:426)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpWaitingBuses(MessageBusImpl.java:387)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:376)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:364)
	at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:357)
	at com.intellij.util.messages.impl.MessageBusImpl.access$200(MessageBusImpl.java:43)
	at com.intellij.util.messages.impl.MessageBusImpl$2.invoke(MessageBusImpl.java:208)
	at com.sun.proxy.$Proxy117.fileOpened(Unknown Source)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.lambda$null$8(FileEditorManagerImpl.java:954)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl$5$1.run(FileEditorManagerImpl.java:1033)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.lambda$doWhenFocusSettlesDown$2(FocusManagerImpl.java:199)
	at com.intellij.util.ui.UIUtil.invokeLaterIfNeeded(UIUtil.java:3094)
	at com.intellij.ide.IdeEventQueue.ifFocusEventsInTheQueue(IdeEventQueue.java:183)
	at com.intellij.ide.IdeEventQueue.executeWhenAllFocusEventsLeftTheQueue(IdeEventQueue.java:132)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:190)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:196)
	at com.intellij.openapi.wm.impl.IdeFocusManagerImpl.doWhenFocusSettlesDown(IdeFocusManagerImpl.java:63)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl$5.run(FileEditorManagerImpl.java:1030)
	at com.intellij.openapi.util.BusyObject$Impl$Simple.execute(BusyObject.java:108)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.notifyPublisher(FileEditorManagerImpl.java:1026)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.lambda$openFileImpl4$9(FileEditorManagerImpl.java:951)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:652)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:665)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.openFileImpl4(FileEditorManagerImpl.java:960)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters$UIBuilder.processFiles(EditorsSplitters.java:844)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters$UIBuilder.processFiles(EditorsSplitters.java:809)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters$ConfigTreeReader.process(EditorsSplitters.java:800)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters.openFiles(EditorsSplitters.java:225)
	at com.intellij.tasks.context.OpenEditorsContextProvider.loadContext(OpenEditorsContextProvider.java:65)
	at com.intellij.tasks.context.WorkingContextManager.loadContext(WorkingContextManager.java:80)
	at com.intellij.tasks.context.WorkingContextManager.lambda$loadContext$2(WorkingContextManager.java:187)
	at com.intellij.tasks.context.WorkingContextManager.doEntryAction(WorkingContextManager.java:197)
	at com.intellij.tasks.context.WorkingContextManager.loadContext(WorkingContextManager.java:185)
	at com.intellij.tasks.context.WorkingContextManager.loadContext(WorkingContextManager.java:242)
	at com.intellij.tasks.context.BranchContextTracker.switchContext(BranchContextTracker.java:61)
	at com.intellij.tasks.context.BranchContextTracker.lambda$branchHasChanged$1(BranchContextTracker.java:56)
	at com.intellij.openapi.application.TransactionGuardImpl.runSyncTransaction(TransactionGuardImpl.java:88)
	at com.intellij.openapi.application.TransactionGuardImpl.lambda$submitTransaction$1(TransactionGuardImpl.java:111)
	at com.intellij.openapi.application.TransactionGuardImpl.submitTransaction(TransactionGuardImpl.java:120)
	at com.intellij.openapi.application.TransactionGuard.submitTransaction(TransactionGuard.java:122)
	at com.intellij.tasks.context.BranchContextTracker.branchHasChanged(BranchContextTracker.java:56)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:117)
	at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:426)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpWaitingBuses(MessageBusImpl.java:387)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:376)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:364)
	at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:357)
	at com.intellij.util.messages.impl.MessageBusImpl.access$200(MessageBusImpl.java:43)
	at com.intellij.util.messages.impl.MessageBusImpl$2.invoke(MessageBusImpl.java:208)
	at com.sun.proxy.$Proxy182.branchHasChanged(Unknown Source)
	at git4idea.branch.GitBranchOperation.lambda$updateRecentBranch$2(GitBranchOperation.java:252)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:315)
	at com.intellij.openapi.application.impl.LaterInvocator$1.run(LaterInvocator.java:156)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(LaterInvocator.java:447)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:431)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:415)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:781)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:722)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:382)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```
![2018-10-31 17_58_33-](https://user-images.githubusercontent.com/14893138/47801347-afab1f00-dd36-11e8-9e67-511192a8cafb.png)

**Comment by zielu** [comment]
@intr Do you have any symlinks in that project by any chance?

**Comment by zielu** [comment]
Fixed in 182.6.2
